openedx/features/survey_report/queries.py

```diff
@@ -0,0 +1,58 @@
+"""
+Queries to get data from database.
+"""
+
+from datetime import datetime, timedelta
+
+from django.contrib.auth.models import User  # lint-amnesty, pylint: disable=imported-auth-user
+from django.db.models import Count, OuterRef, Q, Subquery
+
+from common.djangoapps.student.models import CourseEnrollment
+from lms.djangoapps.grades.models import PersistentCourseGrade
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+
+def get_unique_courses_offered() -> int:
+    """
+    Get total number of unique courses offered.
+    """
+    return CourseOverview.objects.annotate(
+        count=Subquery(
+            CourseEnrollment.objects
+                            .filter(course_id=OuterRef('id'))
+                            .values('course_id')
+                            .annotate(count=Count('course_id'))
+                            .values('count')
```

Comment:
Could you please post the query plan for the generated query?

Reply:
@ormsbee We created this Django query according to thee query sent to the community in the first survey.

```sql
select
    count(*)
from
    course_overviews_courseoverview
where
    start < now()
    and (end > now()
    OR
end is NULL
AND (
    select
        count(*)
    from
        student_courseenrollment
    where
        student_courseenrollment.course_id = course_overviews_courseoverview.id
) > 5
)
```

And we want to get all active courses where the number of enrollments is greater than 5.

Reply:
What's the output when you run an EXPLAIN on this from the mysql prompt, i.e.:

```
explain select
    count(*)
from
    course_overviews_courseoverview
where
    start < now()
    and (end > now()
    OR
end is NULL
AND (
    select
        count(*)
    from
        student_courseenrollment
    where
        student_courseenrollment.course_id = course_overviews_courseoverview.id
) > 5
) \G
 ```

Reply:
@ormsbee 
|id |select_type       |table                          |partitions|type|possible_keys                    |key                              |key_len|ref                                      |rows |filtered|Extra      |
|---|------------------|-------------------------------|----------|----|---------------------------------|---------------------------------|-------|-----------------------------------------|-----|--------|-----------|
|1  |PRIMARY           |course_overviews_courseoverview|          |ALL |                                 |                                 |       |                                         |31014|13.33   |Using where|
|2  |DEPENDENT SUBQUERY|student_courseenrollment       |          |ref |student_courseenrollment_ff48d8e5|student_courseenrollment_ff48d8e5|767    |edxapp.course_overviews_courseoverview.id|14   |100.0   |Using index|